google/cloud/opentelemetry/internal/recordable_test.cc

```diff
@@ -242,6 +270,30 @@ TEST(Recordable, SetAttributeRespectsLimit) {
   EXPECT_EQ(attributes.dropped_attributes_count(), 1);
 }
 
+TEST(Recordable, SetStartTime) {
+  auto start = std::chrono::system_clock::now();
+  auto expected = absl::FromChrono(start);
+
+  auto rec = Recordable(Project(kProjectId));
+  rec.SetStartTime(start);
```

Comment:
For what it's worth, it looks like `opentelemetry::common::SystemTimestamp` has nanosecond resolution, as does `absl::Time` and `google::protobuf::Timestamp`, but `std::chrono::system_clock::time_point` may not on some platforms (notably Windows and macOS), so you should be wary of conversions through `time_point`.

Reply:
Ah.. thanks for investigating. TIL about `system_clock::time_point` not having ns resolution.

I was converting through `chrono`, using this: https://github.com/open-telemetry/opentelemetry-cpp/blob/7cb7654552d68936d70986bc2ee67f3cc3e0b469/api/include/opentelemetry/common/timestamp.h#L53

I updated the `SetStartTime` implementation to convert through `absl::Time`. Do you mind taking another look?